cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -653,8 +661,17 @@ private void assertConsistentTypes(Class<? extends Mapper> firstMapperClass,
       // A bit hacky for stream.
       if (provider instanceof StreamInputFormatProvider) {
         // pass in mapperInput.getMapper() instead of mapperClass, because mapperClass defaults to the Identity Mapper
-        setDecoderForStream((StreamInputFormatProvider) provider, job, inputFormatConfiguration,
-                            mapperInput.getMapper());
+        StreamInputFormatProvider inputFormatProvider = (StreamInputFormatProvider) provider;
+        setDecoderForStream(inputFormatProvider, job, inputFormatConfiguration, mapperInput.getMapper());
+        // Check if the MR job has read access to the stream, if not fail right away. Note that this is being done
```

Comment:
did we change the approach? I thought the intent was that we did not want to capture lineage/usage if the program did not actually write to the stream. 

I sort of agree though that what you have right now is perhaps the right behavior, from a usage point of view. Usage should not care if a principal has access to a dataset, when a program accesses a dataset. I am not completely sure about lineage though, since lineage should tell me if a dataset was influenced by an operation (READ or WRITE) in a given time range. So it seems like if authorization fails, lineage should not be captured.

What do you think?